### ISSUE TITLE
Clarify multi-threading support in C implementation

### ISSUE THREAD

**Issue by k0001** [BLAKE3-team/BLAKE3#99]
The `README` in the C implementation says:

> The C implementation doesn't currently support multi-threading. 

Could you clarify which of the following do you mean by that?

1. That the C implementation doesn't do any multi-threading on itself.

2. That it is not safe to use the C implementation as is on multi-threaded environments. That is, one must not use the exported C functions concurrently.

Thank you.

**Comment by oconnor663** [comment]
Good point, we should make this explicit in the docs. We mean number 1. The C implementation doesn't do any multi-threading optimizations internally. But it's safe to use BLAKE3 from multiple threads, as long as each thread is operating on its own separate `blake3_hasher` object.

**Comment by arvidn** [comment]
blake3 does not support (2) either right now. There is a data race with the global variable `g_cpu_features`.
I've posted an issue here: https://github.com/BLAKE3-team/BLAKE3/issues/220
